snafu/process.py

```diff
@@ -30,152 +28,7 @@ class ProcessSample:
-
-        self._cleaned: bool = False
-        self._stdout: bytes = b""
-        self._stderr: bytes = b""
-        self._threads: Optional[List[threading.Thread]] = None
-
-        self._check_pipes(self.kwargs)
-
-    @staticmethod
-    def _check_pipes(kwargs):
-        if (
-            not kwargs.get("stdout", False)
-            and not kwargs.get("stderr", False)
-            and not kwargs.get("capture_output", False)
-        ):
-            kwargs["stdout"] = subprocess.PIPE
-            kwargs["stderr"] = subprocess.PIPE
-
-    def _enqueue_line_from_fh(self, file_handler, queue_attr, store):
-        if file_handler is not None:
-            for line in iter(file_handler.readline, b""):
-                queue_attr.put(line)
-                # use this method since running in separate thread
-                setattr(self, store, getattr(self, store) + line)
-
-    def start(self):
-        """Start the subprocess and create threads for capturing output."""
-
-        self.start_time = datetime.datetime.utcnow()
-        self.process = subprocess.Popen(self.cmd, **self.kwargs)  # pylint: disable=R1732
-        self._threads = [
-            threading.Thread(
-                target=self._enqueue_line_from_fh,
-                args=(self.process.stdout, self.stdout, "_stdout"),
-                daemon=True,
-            ),
-            threading.Thread(
-                target=self._enqueue_line_from_fh,
-                args=(self.process.stderr, self.stderr, "_stderr"),
-                daemon=True,
-            ),
-        ]
-
-        for thread in self._threads:
-            thread.start()
-
-    def __enter__(self):
-        """Call start method and return self."""
-        self.start()
-        return self
-
-    def cleanup(self):
-        """Cleanup the subprocess with a timeout and populate the ProcessRun instance at ``attempt``."""
-        if not self._cleaned:
-            if self.timeout is not None:
-                try:
-                    self.process.wait(timeout=self.timeout)
-                    self.attempt.hit_timeout = False
-                except subprocess.TimeoutExpired:
-                    self.attempt.hit_timeout = True
-                    self.process.kill()
-                    self.process.wait()
-            else:
-                self.process.wait()
-
-            self.end_time = datetime.datetime.utcnow()
-            for thread in self._threads:
-                thread.join()
-
-            self._cleaned = True
-            self.attempt.stdout = self._stdout.decode("utf-8")
-            self.attempt.stderr = self._stderr.decode("utf-8")
-            self.attempt.rc = self.process.returncode
-            self.attempt.time_seconds = (self.end_time - self.start_time).total_seconds()
-
-    def __exit__(self, *args, **kwargs):
-        """Call cleanup method on exit of context."""
-        self.cleanup()
+    successful: Optional[ProcessRun] = None
 
 
 def get_process_sample(
```

Comment:
Remove mention of LiveProcess in doc string.
```
    """
    Run the given command as a subprocess, retrying if the subprocess fails, returning a
    :py:class:`~snafu.process.ProcessSample` detailing the results.

    This function expects a logger because it is expected that it will be used by benchmarks, which should
    be logging their progress anyways.

    Parameters
    ----------
    cmd : str or list of str
        Command to run. Can be string or list of strings if using :py:mod:`shlex`
    logger : logging.Logger
        Logger to use in order to log progress.
    retries : int, optional
        Number of retries to perform. Defaults to zero, which means that the function will run the process
        once, not retrying on failure.
    expected_rc : int, optional
        Expected return code of the process. Will be used to determine if the process ran successfully or not.
    timeout : int, optional
        Time in seconds to wait for process to complete before killing it.
    kwargs
        Extra kwargs will be passed to :py:class:`~snafu.process.ProcessSample`

    Returns
    -------
    ProcessSample
    """
```

Reply:
Thank you for the catch!